### PR TITLE
fix for gcc12 warnings

### PR DESCRIPTION
**Pull request by samuelpmishLLNL**
includes a fix for some of the warnings/errors that showed up when compiling with gcc-12:

```
[build]     inlined from ‘serac::BoundaryCondition::vectorCoefficient()’ at /home/sam/code/serac/src/serac/physics/boundary_conditions/boundary_condition.cpp:147:12:
[build] /usr/include/c++/12/bits/shared_ptr_base.h:1666:16: error: potential null pointer dereference [-Werror=null-dereference]
[build]  1666 |       { return _M_ptr; }
```

as mentioned in https://github.com/LLNL/serac/issues/848 

Also disables a few of the legacy tests that were failing due to a DSuperLU segfault.

### CODE REVIEW

**Review by jamiebramwell — APPROVED** [review]

File: src/serac/physics/boundary_conditions/boundary_condition.cpp [hunk 1]
```diff
@@ -118,33 +118,45 @@ void BoundaryCondition::apply(mfem::HypreParMatrix& k_mat, mfem::Vector& rhs, mf
...
 {
   auto scalar_coef = get_if<std::shared_ptr<mfem::Coefficient>>(&coef_);
-  SLIC_ERROR_ROOT_IF(!scalar_coef,
-                     "Asking for a scalar coefficient on a BoundaryCondition that contains a vector coefficient.");
-  return **scalar_coef;
+  if (scalar_coef) {
+    return **scalar_coef;
+  } else {
+    SLIC_ERROR_ROOT("Asking for a scalar coefficient on a BoundaryCondition that contains a vector coefficient.");
+    exit(1);
```
Comment (white238) [inline src/serac/physics/boundary_conditions/boundary_condition.cpp:125]: You don't need the `exit(1)` here. SLIC_ERROR_ROOT handles this.

Comment (white238) [inline src/serac/physics/boundary_conditions/boundary_condition.cpp:125]:
```suggestion
```

File: src/serac/physics/boundary_conditions/boundary_condition.cpp [hunk 2]
```diff
@@ -118,33 +118,45 @@ void BoundaryCondition::apply(mfem::HypreParMatrix& k_mat, mfem::Vector& rhs, mf
...
```
Comment (white238) [inline src/serac/physics/boundary_conditions/boundary_condition.cpp:136]:
```suggestion
```

File: src/serac/physics/boundary_conditions/boundary_condition.cpp [hunk 3]
```diff
@@ -118,33 +118,45 @@ void BoundaryCondition::apply(mfem::HypreParMatrix& k_mat, mfem::Vector& rhs, mf
...
 {
   auto vec_coef = get_if<std::shared_ptr<mfem::VectorCoefficient>>(&coef_);
-  SLIC_ERROR_ROOT_IF(!vec_coef,
-                     "Asking for a vector coefficient on a BoundaryCondition that contains a scalar coefficient.");
-  return **vec_coef;
+  if (vec_coef) {
+    return **vec_coef;
+  } else {
+    SLIC_ERROR_ROOT("Asking for a vector coefficient on a BoundaryCondition that contains a scalar coefficient.");
+    exit(1);
```
Comment (white238) [inline src/serac/physics/boundary_conditions/boundary_condition.cpp:147]:
```suggestion
```

File: src/serac/physics/boundary_conditions/boundary_condition.cpp [hunk 4]
```diff
@@ -118,33 +118,45 @@ void BoundaryCondition::apply(mfem::HypreParMatrix& k_mat, mfem::Vector& rhs, mf
...
```
Comment (white238) [inline src/serac/physics/boundary_conditions/boundary_condition.cpp:158]:
```suggestion
```

**Review by white238 — APPROVED** [review]

File: src/serac/physics/boundary_conditions/boundary_condition.cpp (hunk 1 above)
Comment (white238) [reply src/serac/physics/boundary_conditions/boundary_condition.cpp:125]: Jamie explained its the compiler warning there isn't a return/exit from this case. I'll log a SLIC bug.

**Pull request merged**